lib/cloud/clients.go

```diff
@@ -942,3 +1029,108 @@ func (c *TestCloudClients) GetAzureRunCommandClient(subscription string) (azure.
 func (c *TestCloudClients) Close() error {
 	return nil
 }
+
+// AWSSessionCacheKeyBuilder is a helper struct wrapper around a string builder
+// that is used to efficiently build cache keys for a given region and, optionally,
+// a chain of AWS assumed-role sessions.
+type AWSSessionCacheKeyBuilder struct {
+	sb        *strings.Builder
+	roleCount int
+}
+
+// NewAWSSessionCacheKeyBuilder constructs a new cache key builder intialized
```

Comment:
```suggestion
// NewAWSSessionCacheKeyBuilder constructs a new cache key builder initialized
```

Reply:
obsolete, removed this cache key builder in fc800e15f75f7ed8f38b88b6aba00a756c66d144